### ISSUE TITLE
Email logo not displaying on Return Emails

### ISSUE THREAD

**Issue by vivekjha25** [magento/magento2#19581]
### Preconditions (*)

1. Magento 2.2.6
2. 

### Steps to reproduce (*)

1. Set a logo for transactional emails.
2. Create return for a completed order
3. Check the logo in the return mail

### Expected result (*)

Logo should be displayed as expected.

### Actual result (*)

Logo not displaying as expected

**Comment by magento-engcom-team** [comment]
Hi @vivekjha25. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@vivekjha25 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96972`, `MAGETWO-96973` were created

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by webinti** [comment]
Do you have a fix for that please?

**Comment by webusat** [comment]
I have the same problem with a fresh install of 2.3.  I click Upload to upload the logo image, choose the image and I get a running circle for a second and nothing is ever added.  No logo image is displayed in the backend at all and the image in the email is still the default Magento logo.

**Comment by CompactCodeEU** [comment]
### **OLD COMMENT (See edit section in this answer)**

Note that i am writing this answer as i am debugging.

The same issue persists here without an error notifcation or a apache error or any magento error log. Not even in developer mode. Kind of hard to debug like this ofcourse.

It does an ajax call to its own page (theme/design_config/edit) and not to the fileuploader so it makes sense. The ajax call also returns a full page as a response.

This form still uses the formElement FileUploader but the it has the new tag of imageUploader under `<formElements>` .0 (i can know because we use them in our theme) in `vendor/magento/module-email/view/adminhtml/ui_component/design_config_form.xml` Line 16.


```
        <field name="email_logo" formElement="fileUploader">
                <settings>
                    <notice translate="true">To optimize logo for high-resolution displays, upload an image that is 3x normal size and then specify 1x dimensions in the width/height fields below.</notice>
                    <label translate="true">Logo Image</label>
                    <componentType>imageUploader</componentType>
                </settings>
                <formElements>
                    <imageUploader>
                        <settings>
                            <allowedExtensions>jpg jpeg gif png</allowedExtensions>
                            <maxFileSize>2097152</maxFileSize>
                            <uploaderConfig>
                                <param xsi:type="string" name="url">theme/design_config_fileUploader/save</param>
                            </uploaderConfig>
                        </settings>
                    </imageUploader>
                </formElements>
            </field>
```
Change that to

```
        <field name="email_logo" formElement="fileUploader">
                <settings>
                    <notice translate="true">To optimize logo for high-resolution displays, upload an image that is 3x normal size and then specify 1x dimensions in the width/height fields below.</notice>
                    <label translate="true">Logo Image</label>
                    <componentType>imageUploader</componentType>
                </settings>
                <formElements>
                    <fileUploader>
                        <settings>
                            <allowedExtensions>jpg jpeg gif png</allowedExtensions>
                            <maxFileSize>2097152</maxFileSize>
                            <uploaderConfig>
                                <param xsi:type="string" name="url">theme/design_config_fileUploader/save</param>
                            </uploaderConfig>
                        </settings>
                    </fileUploader>
                </formElements>
            </field>
```
and it works again.

The thing is that i think a partial update has been done here. Since 2.3.0 (maybe sooner not sure) there is an imageUploader instead of a fileUploader. We do not use that because it is not compatible with older versions of Magento but you can use it like this (i know this example has a mix of the data/config and settings tags but its the quickest example that i have at the current time)


     <field name="email_logo" formElement="imageUploader">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="source" xsi:type="string">image</item>
                    <item name="visible" xsi:type="boolean">true</item>
                    <item name="notice" xsi:type="string" translate="true">To optimize logo for high-resolution displays, upload an image that is 3x normal size and then specify 1x dimensions in the width/height fields below.</item>
                </item>
            </argument>
            <settings>
                <elementTmpl>ui/form/element/uploader/uploader</elementTmpl>
                <dataType>Logo Image</dataType>
                <label translate="true">Image</label>
            </settings>
            <formElements>
                <imageUploader>
                    <settings>
                        <uploaderConfig>
                            <param xsi:type="url" name="url" path="theme/design_config_fileUploader/save"/>
                        </uploaderConfig>
                    </settings>
                </imageUploader>
            </formElements>
        </field>

which will give you the new fancy "select from gallery" function

![image](https://user-images.githubusercontent.com/41287340/52140216-76928b80-2652-11e9-9300-da16c811e2bd.png)


Not sure how to limit the number of images that you can upload in that one since. I tried                         `<item name="isMultipleFiles" xsi:type="string">false</item>` but without any success.

Anyways i will make a request to make the changes


###  **EDIT** 

There is already a patch for this and i missed something
You should just change 
`        <field name="email_logo" formElement="fileUploader">`
to
`            <field name="email_logo" formElement="imageUploader">
`

**References1 :** 

https://magento.stackexchange.com/questions/256668/magento-2-3-0-transactional-email-logo-image-upload-not-working

**References 2 :** 

https://github.com/magento/magento2/issues/18688

**Comment by chirag-d-wagento** [comment]
I am working on this issue at #MM19IN

**Comment by chirag-d-wagento** [comment]
I am working on this issue at #MM19IN

**Comment by magento-engcom-team** [comment]
@chirag-d-wagento thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @chirag-d-wagento. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by davidverholen** [comment]
already fixed in 2.2 and 2.3
see https://github.com/magento/magento2/issues/18688
